src/ssh-compute.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+
+import path from 'path';
+import { promises as fs } from 'fs';
+
+import { SSH_PUBLIC_KEY_FILENAME, SSH_PRIVATE_KEY_FILENAME } from './const';
+
+export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
+export const GCLOUD_METRICS_LABEL = 'github-actions-ssh-compute';
+
+export function parseFlags(flags: string): RegExpMatchArray {
+  return flags.match(/(".*?"|[^"\s=]+)+(?=\s*|\s*$)/g)!; // Split on space or "=" if not in quotes
+}
+
+/**
+ * Executes the main action. It includes the main business logic and is the
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  core.exportVariable(GCLOUD_METRICS_ENV_VAR, GCLOUD_METRICS_LABEL);
+  try {
```

Comment:
It's probably worth moving the `try-catch` into `main` to simplify this indentation a bit.